### DISCUSSION TITLE
popTo with nested navigation

### DISCUSSION THREAD

**Discussion by brmenchl** [react-navigation/react-navigation#12072] [Q&A] [answered]
If I want to pop to a nested screen, how would I do it? When I call

`navigation.popTo('Screen', { screen: 'SubScreen' });`

the screen is pushed onto the stack, even if it exists in history. I tried adding `getId` to the screens. This prevents an extra instance of the screen from being pushed, but still pulls that screen to the top of the stack, rather than popping to it.

I did notice in a similar question, there was a suggestion to target a specific navigator using key. I thought key was being deprecated from the public API in 7, and getId was suggested instead. It feels pretty brittle to search through the nav state for the key of the stack route in order to supply it to the popTo action

**Comment by satya164** [comment, marked as answer]
The `.popTo` applies to the screen you're navigating to, not the nested screen you pass in params. The screen passed in params will always result in a `navigate`.

Passing route `key` in the action is removed. But the `target` is supported to target a specific navigator. See https://reactnavigation.org/docs/7.x/navigation-object/#dispatch

Alternatively you can:
- dispatch an event using an event emitter library and have a screen inside your stack listen to the event to do an appropriate action
- use the `reset` action where you modify the state of the child navigator

Note that it's upto you to make sure that the navigator is already rendered for any of this to work.

The easiest way to do it to it would be to dispatch `.popTo` from the child navigator itself and not from a parent. So I'd recommend reworking your logic to do that instead. Generally dispatching actions from parent to child will be tricky.

**Reply by brmenchl** [reply]
I see, ok. I think we can reorganize some of our app to reduce nesting, which may fix this issue. Thanks for the response!
